### PR TITLE
updated final report

### PR DESCRIPTION
**Pull request by katieb1**

### CODE REVIEW

**Review by andrealee011 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ Rscript -e "rmarkdown::render('src/DSCI_522_EDA.Rmd')"
 Rscript src/prepare_data.R --datafile=data/avocado.csv --out=data
 Rscript src/render_EDA.R --datafile=data/train.feather --out=doc/img
 Rscript src/conduct_hypothesis_test.R --datafile=data/train.feather --out=doc/img
+python src/regression.py data/train.feather results/
```
Comment (techrah) [inline README.md:48]: What's the command for the final report that we run from the command line?

Comment (katieb1) [reply README.md:48]: I will add it in and then re-push
I think it should be:
`Rscript -e "rmarkdown::render('doc/avocado_predictors_report.Rmd', output_format = 'github_document')"`

File: doc/avocado_predictors_report.Rmd [hunk 2]
```diff
@@ -82,49 +83,72 @@ However, region and month are categorical variables that have numerous levels. T
...
+The high VIF scores suggest extremely high collinearity for these variables in a linear model. Therefore, we will be careful about using these features as they are probably ineffective predictors of the average avocado price.
 
 ## Random Forest Feature Importances
 
-Lastly, we fitted a random forest regressor model using the features that we determined as significant from the analysis above (`region`, `type`, and `month`). We used one hot encoding on these categorical features and used cross validation to determine the optimal hyperparameters. We found that our top predictor of avocado prices is `type` (i.e. whether the avocado is organic or convenional).
+Lastly, we fitted a random forest regressor model using the features that we determined as significant from the analysis above (`region`, `type`, and `month`). We used one hot encoding on these categorical features and used randomized cross validation to determine the optimal hyperparameters, maximum depth and number of estimators. We calculated the cross-validation scores to determine how well our model was performing.
```
Comment (techrah) [inline doc/avocado_predictors_report.Rmd]: Hmm, can we say "cross-validation" scores? How about "We calculated the average (validation) scores using cross validation..."

Comment (katieb1) [reply doc/avocado_predictors_report.Rmd]: I like this! I will change it.

Comment (techrah) [reply doc/avocado_predictors_report.Rmd]: I see you used this terminology in other places too. If that's how it's used, I'm ok with leaving as is. I was just curious as I've not come across it worded like that before.

File: doc/avocado_predictors_report.Rmd [hunk 3]
```diff
@@ -39,27 +41,26 @@ The features we tested were:
...
-- `year`: year avocado was sold in
 - `region`: U.S. state avocado was sold in
 - `month`: month avocado was sold in
 
 The target was:
   
 - `average_price`: average price of avocado sold
 
-To perform this anlaysis, the R and Python programming languages [@r; @Python] as well as the following R and Python packages were used: caret [@caret], docopt [@docopt], feather [@feather], tidyverse [@tidyverse], lubridate [@lubridate], corr [@corr], car [@car], kableExtra [@kableExtra], reshape2 [@reshape2], tidyverse [@tidyverse], broom [@broom], knitr [@knitr], ADD PACKAGES USED FOR ML.
+To perform this anlaysis, the R and Python programming languages [@r; @Python] as well as the following R and Python packages were used: caret [@caret], docopt [@docopt], feather [@feather], tidyverse [@tidyverse], lubridate [@lubridate], corr [@corr], car [@car], kableExtra [@kableExtra], reshape2 [@reshape2], tidyverse [@tidyverse], broom [@broom], knitr [@knitr], pandas [@pandas], numpy[@numpy], scikit-learn[@scikit-learn], and altair[@2018-altair].
```
Comment (techrah) [inline doc/avocado_predictors_report.Rmd]: I also had to install the `pyarrow` and `selenium` Python packages as well as `chromedriver` at the OS level in order to get the regression script to run successfully.

Comment (techrah) [reply doc/avocado_predictors_report.Rmd]: We also need to add the `here` R package.

File: README.md (hunk 1 above)
Comment (techrah) [reply README.md:48]: Great. Thanks!

File: doc/avocado_predictors_report.Rmd (hunk 3 above)
Comment (katieb1) [reply doc/avocado_predictors_report.Rmd]: ok I will add those ones as well

Comment (katieb1) [reply doc/avocado_predictors_report.Rmd]: I can't find the bibtex for chromedriver or pyarrow

Comment (katieb1) [reply doc/avocado_predictors_report.Rmd]: Why don't we just merge what we have for now, and then you guys can work on the final report as well and I can see if I can find those references

Comment (andrealee011) [reply doc/avocado_predictors_report.Rmd]: i tried looking and i can't find it either...

Comment (katieb1) [reply doc/avocado_predictors_report.Rmd]: added here package

Comment (techrah) [reply doc/avocado_predictors_report.Rmd]: ok, sounds good.

**Pull request merged**